app/code/Magento/ProductAlert/Controller/Unsubscribe/Stock.php

```diff
@@ -21,16 +23,33 @@ class Stock extends UnsubscribeController
     protected $productRepository;
 
     /**
-     * @param \Magento\Framework\App\Action\Context $context
-     * @param \Magento\Customer\Model\Session $customerSession
-     * @param \Magento\Catalog\Api\ProductRepositoryInterface $productRepository
+     * @var Magento\Store\Model\StoreManagerInterface
+     */
+    protected $storeManager;
+
+    /**
+     * @var \Magento\ProductAlert\Model\StockFactory
+     */
+    protected $stockFactory;
+
+    /**
+     * Stock constructor.
+     * @param Context $context
+     * @param CustomerSession $customerSession
+     * @param ProductRepositoryInterface $productRepository
+     * @param StoreManagerInterface $storeManager
+     * @param \Magento\ProductAlert\Model\StockFactory $stockFactory
```

Comment:
Please use short name of the class.

Reply:
not used due to same name was already used in current class.

Reply:
Probably, it can be caused that class already imported and only you need - simplify FQN and leave only class name.
Also, in similar situations, you can always use aliases for the imported class name.
It is more clear to keep all class names in the class are imported or not imported at all.

However, I can not see 'use' section in changes. Has it occasionally removed with force push?

Reply:
you are right ...added stockFactory.